src/software/new_geom/polynomial2d.h

```diff
@@ -108,3 +106,25 @@ Polynomial2d operator+(const Polynomial2d &p1, const Polynomial2d &p2);
  * @return the new Polynomial2d
  */
 Polynomial2d &operator+=(Polynomial2d &p1, const Polynomial2d &p2);
+
+/**
+ * Subtracts a Polynomial2d to another Polynomial2d
+ *
+ * @param p1 the Polynomial2d to subtract from.
+ * @param p2 the Polynomial2d to subtract.
+ *
+ * @return the new Polynomial2d
+ */
+Polynomial2d &operator-=(Polynomial2d &p1, const Polynomial2d &p2);
```

Comment:
Can we change all of these to return a `Polynomial2d` not a `Polynomial2d&`. In most cases, we're return a reference to a stack object